This branch is to Refactor script file via export & import to modules and update DateTime through luxon package

steveWDamesJr opened the pull request:
Refactor:

-Constructor | Store | UI classes into modules/ import files into index.js
-Toggle code into es6 arrow functions
-Update date with Luxon package DateTime
-Debug Linters

steveWDamesJr committed:
Initialize build

Add Files:
-README.md: Update Readme to project specs
-.gitignore: Update with files to ignore
-index.js, index.css: connect script and stylesheet to markup
-Modules: add modules directory
-package.json

steveWDamesJr committed:
Update linters

steveWDamesJr committed:
Refactor scripts

-Export and Import classes between modules and index.js
-Add luxon package to import DateTime functionality
-Linter checks

steveWDamesJr committed:
Fix linter errors

steveWDamesJr committed:
Fix linter errors

steveWDamesJr committed:
Fix linter errors

steveWDamesJr committed:
Fix linter errors

steveWDamesJr committed:
Fix linter errors

steveWDamesJr committed:
Fix linter errors

steveWDamesJr committed:
Fix linter errors

steveWDamesJr committed:
Fix linter errors

steveWDamesJr committed:
Fix linter errors

steveWDamesJr committed:
Fix linter errors

Mihndim2020 commented on README.md:
```diff
@@ -1,59 +1,53 @@
 Additional description about the project and its features.
 
+# Screenshots
+![DESKTOP VIEW](./images/awesome-books.png)
+![MOBILE VIEW](./images/awesome-books-mobile.png)
+![DROP DOWN MENU MOBILE VIEW](./images/awesome-books-drop-down.png)
```
- [x] Please the links to the screenshots are broken, kindly fix that as this will make your `readme file` look more professional.

Mihndim2020 commented on index.css:
```diff
@@ -0,0 +1,166 @@
+/* stylelint-disable-next-line no-empty-first-line */
```
- [x] `Linters` are tools that perform static analysis on software in order to recognize and report adherence/non-adherence to some set of coding best practice.Instead of disabling `stylelint`, it will be nice to fix the issue.

Mihndim2020 commented on index.js:
```diff
@@ -0,0 +1,74 @@
+import UI from './modules/ui.js';
+import Store from './modules/store.js';
+
+// eslint-disable-next-line max-classes-per-file
```
- [x] `Linters` are tools that perform static analysis on software in order to recognize and report adherence/non-adherence to some set of coding best practice.Instead of disabling `eslint`, it will be nice to fix the issue.

Mihndim2020 commented on modules/ui.js:
```diff
@@ -0,0 +1,36 @@
+
+export default class UI {
+  static displayBooks() {
+    // eslint-disable-next-line no-use-before-define
```
- [x] `Linters` are tools that perform static analysis on software in order to recognize and report adherence/non-adherence to some set of coding best practice.Instead of disabling `eslint`, it will be nice to fix the issue.

Mihndim2020 commented on index.js:
```diff
@@ -0,0 +1,74 @@
+import Store from './modules/store.js';
+
+// eslint-disable-next-line max-classes-per-file
+const { luxon } = window;
+const dt = luxon.DateTime.now();
```
- [ ] As it is a requirement for this project, you are suppose to display the date using `luxon` library. Please check the screenshot of the requirements below and kindly fix that.
![luxon2](https://user-images.githubusercontent.com/65852621/149945930-14db171f-a757-4cd4-85f3-f1ea3a0a66c9.png)

Mihndim2020 commented on modules/store.js:
```diff
@@ -0,0 +1,32 @@
+    return books;
+  }
+
+  static addBook(book) {
+    const books = Store.getBooks();
+
+    books.push(book);
+
+    localStorage.setItem('books', JSON.stringify(books));
+  }
+
+  static removeBook(isbn) {
+    const books = Store.getBooks();
+
+    books.forEach((book, index) => {
+      if (book.isbn === isbn) {
+        books.splice(index, 1);
+      }
+    });
+
+    localStorage.setItem('books', JSON.stringify(books));
```
- [x] As it is a requirement for this project, please make sure your app is working and it is possible to add and remove books from the list. For now, I am unable to `add` and `remove` books.

Mihndim2020 reviewed (changes requested):
### Hi @, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  No `Linter` errors,
:heavy_check_mark: Your project is running. 

There are some issues that you still need to work on to get this project approved, but you are almost there!
![ALMOST-MINION](https://user-images.githubusercontent.com/65852621/133680327-431cb21d-ad7c-483a-84eb-21b733980a04.png)

# Status: Requires Changes ♻️
_Check the comments below and under the review._

### I will leave the comment here, since it concerns files in your repository ###

- [ ] Your project should contain a `package.json` file which records important metadata about a project which is required before publishing to NPM, and also defines functional attributes of a project that npm uses to install dependencies, run scripts, and identify the entry point to our package. Checkout the screenshot of the requirements below and kindly fix the issue. 
![package json](https://user-images.githubusercontent.com/65852621/149948016-cbad9a4a-77a2-4aa1-a9ea-d5a70029507e.png)

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

steveWDamesJr committed:
Fix errors

steveWDamesJr committed:
Fix errors

steveWDamesJr committed:
Fix errors

steveWDamesJr committed:
Fix errors

steveWDamesJr committed:
Fix errors

steveWDamesJr committed:
Fix errors

steveWDamesJr committed:
Fix required changes

-Add Package.json
-Fix broken links in README
-Fix disabled stylelint error
-Fix disabled eslint errors
-DateTie Package imported from Luxon library

steveWDamesJr committed:
Fix errors

steveWDamesJr committed:
Update README

VanessaAoki reviewed (approved):
## STATUS: APPROVED 🟢 
Hi @steveWDamesJr, 

Great job implementing all the changes required by the previous Code Reviewer, your project is complete! There is nothing else to say other than... it's time to merge it! 
Congratulations! 🎉 

<img alt="approved" src="https://64.media.tumblr.com/87e0b6469c957bf896cdec1cab646acf/tumblr_inline_pk1bdakCCp1rpcr9a_500.gif">

### Project highlights:
- [x] All linter checks are passing :heavy_check_mark: 
- [x] Your README file looks very professional and the PR has a descriptive title/summary of the implemented changes :memo: 
- [x] Descriptive commits and correct use of the GitHub Flow :fire: 


Cheers and Happy coding!👏👏👏

- [OPTIONAL]: You should always keep your node modules folder in the gitingore file!! 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
